### PR TITLE
pwndbg: 2022.12.19 -> 2023.07.17

### PR DESCRIPTION
**Pull request by patryk4815**
###### Description of changes

Bump to "2023.07.17" and update dependencies.
And fix error with (setting LC_CTYPE fix this error):
```
  File "/nix/store/jpadwyb4ds4705jm4pzx5snsggl722as-pwndbg-2023.03.19/share/pwndbg/pwndbg/gdblib/config.py", line 46, in __init__
    self.value = param.value
UnicodeEncodeError: 'ascii' codec can't encode characters in position 0-1: ordinal not in range(128)
```

Can be tested :)
```
nix run github:patryk4815/nixpkgs/pwndbg_2023.03.19#pwndbg
```

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [ ] ~~aarch64-darwin~~
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by patryk4815** [comment]
@ofborg build pwndbg

**Comment by patryk4815** [comment]
@ofborg build pwndbg

File: pkgs/development/tools/misc/pwndbg/default.nix [hunk 1]
```diff
@@ -47,6 +50,7 @@ in stdenv.mkDerivation rec {
     makeWrapper ${gdb}/bin/gdb $out/bin/pwndbg \
       --add-flags "-q -x $out/share/pwndbg/gdbinit.py" \
       --prefix PATH : ${binPath} \
+      --set LC_CTYPE "C.UTF-8" \
```
Comment (Mic92) [inline pkgs/development/tools/misc/pwndbg/default.nix:53]: Users might want to set it to something else especially when debugging:
```suggestion
      --set-default LC_CTYPE "C.UTF-8" \
```

Comment (Mic92) [reply pkgs/development/tools/misc/pwndbg/default.nix:53]: Mhm. I don't thing we have C.utf-8 on macOS actually.

Comment (patryk4815) [reply pkgs/development/tools/misc/pwndbg/default.nix:53]: Hmm, yes, on macos we need recompile gdb with custom charset.. :(

Comment (disconnect3d) [reply pkgs/development/tools/misc/pwndbg/default.nix:53]: Hi, Pwndbg maintainer here. We don't really support MacOS well in Pwndbg (hopefully yet), so I think the lack of C.utf-8 on macOS is not an issue here @Mic92.

We should just proceed with merging this and actually further update this to the latest release from yesterday (2023.07.17).

**Comment by patryk4815** [comment]
There is missing `capstone==5.0.0.post1` in nixpkgs for now.   @Mic92 how to handle different version only for this python dependency? In nixpkgs we have right now capstone=4.0.2.

**Comment by SuperSandro2000** [comment]
> There is missing `capstone==5.0.0.post1` in nixpkgs for now (required version used in pwndbg). Mic92 how to handle different version only for this python dependency? In nixpkgs we have right now capstone=4.0.2.

First we usually unpin those because they are a headache to work with and second we should try updating capstone. If that doesn't work we can use a local overwrite.

File: pkgs/development/tools/misc/pwndbg/default.nix [hunk 2]
```diff
@@ -27,14 +30,14 @@ let
...
+  version = "2023.07.17";
   format = "other";
 
   src = fetchFromGitHub {
     owner = "pwndbg";
     repo = "pwndbg";
     rev = version;
-    sha256 = "sha256-pyY2bMasd6GaJZZjLF48SvkKUBw3XfVa0g3Q0LiEi4k=";
+    sha256 = "sha256-HElzkHv94tEVq1mUeNo75Ja7yMaup2CIh96nWivkkjg=";
     fetchSubmodules = true;
```
Comment (SuperSandro2000) [inline pkgs/development/tools/misc/pwndbg/default.nix:41]:
```suggestion
    fetchSubmodules = true;
    hash = "sha256-HElzkHv94tEVq1mUeNo75Ja7yMaup2CIh96nWivkkjg=";
```

File: pkgs/development/tools/misc/pwndbg/default.nix [hunk 3]
```diff
@@ -7,17 +7,20 @@
 }:
 
 let
-  pythonPath = with python3.pkgs; makePythonPath [
+  venv = python3.withPackages (p: with p; [
```
Comment (SuperSandro2000) [inline pkgs/development/tools/misc/pwndbg/default.nix:10]:
```suggestion
  pyEnv = python3.withPackages (ps: with ps; [
```
that's more standard naming

**Comment by patryk4815** [comment]
Waiting for merge: https://github.com/NixOS/nixpkgs/pull/245527 https://github.com/NixOS/nixpkgs/pull/245576

**Comment by patryk4815** [comment]
#302057 is better ;)

**Pull request closed without merging**